app/src/test/java/org/oppia/app/testing/ImageRegionSelectionInteractionViewTest.kt

```diff
@@ -79,81 +98,103 @@ class ImageRegionSelectionAccessibilityTest {
   }
 
   @Test
-  fun testImageRegionSelectionTestActivity_clickOnRegion3() {
+  fun testImageRegionSelectionInteractionView_clickRegion3_Region2Clicked() {
     launch(ImageRegionSelectionTestActivity::class.java).use {
-      onView(withId(R.id.clickable_image_view)).perform(
-        clickAtXY(0.3f, 0.3f)
-      )
-      onView(allOf(withTagValue(`is`("Region 3"))))
-        .check(
-          matches(isDisplayed())
-        )
-      onView(allOf(withId(R.id.submit_button)))
-        .check(
-          matches(isEnabled())
+      it.onActivity {
+        it.clickable_image_view.setListener(onClickableAreaClickedListener)
+
+        onView(withId(R.id.clickable_image_view)).perform(
+          clickPoint(0.2f, 0.2f)
         )
+
+        verify(onClickableAreaClickedListener).onClickableAreaTouched(capture(regionClickedEvent))
+        assertEquals(NamedRegionClickedEvent("Region 2"), regionClickedEvent.value)
+      }
     }
   }
 
   @Test
-  fun testImageRegionSelectionTestActivity_clickOnRegion3_clickOnRegion2_onlyRegion2IsDisplayed() {
+  fun testImageRegionSelectionInteractionView_clickRegion3_clickRegion2_Region2Clicked() {
     launch(ImageRegionSelectionTestActivity::class.java).use {
       onView(withId(R.id.clickable_image_view)).perform(
-        clickAtXY(0.3f, 0.3f)
+        clickPoint(0.3f, 0.3f)
       )
       onView(allOf(withTagValue(`is`("Region 3"))))
         .check(
           matches(isDisplayed())
         )
 
       onView(withId(R.id.clickable_image_view)).perform(
-        clickAtXY(0.7f, 0.3f)
+        clickPoint(0.7f, 0.3f)
       )
       onView(allOf(withTagValue(`is`("Region 2"))))
         .check(
           matches(isDisplayed())
         )
-      onView(allOf(withId(R.id.submit_button)))
-        .check(
-          matches(isEnabled())
-        )
     }
   }
 
   @Test
-  fun testImageRegionSelectionTestActivity_clickOnDefaultRegion() {
+  fun testImageRegionSelectionInteractionView_clickOnDefaultRegion_defaultRegionClicked() {
     launch(ImageRegionSelectionTestActivity::class.java).use {
       onView(withId(R.id.clickable_image_view)).perform(
-        clickAtXY(0.0f, 0.0f)
+        clickPoint(0.0f, 0.0f)
       )
       onView(withId(R.id.default_selected_region)).check(
-        matches((isDisplayed()))
+        matches(isDisplayed())
       )
-      onView(allOf(withId(R.id.submit_button)))
-        .check(
-          matches(not(isEnabled()))
-        )
     }
   }
 
   @Test
-  fun testImageRegionSelectionTestActivity_clickOnDefaultRegionWithAccessibility() {
+  fun testImageRegionSelectionInteractionView_withTalkbackEnabled_clickRegion3_clickRegion2_Region2Clicked() { // ktlint-disable max-line-length
```

Comment:
Prefer to shorten the test name to avoid needing to disable the max line-length check (this check should only be disabled when there's no other choice). E.g.:

``testView_withTalkbackEnabled_clickRegion3_clickRegion2_Region2Clicked``

(note that the long test...View bit in the beginning of the test names isn't necessary since that context is already communicated via the class's name).

Reply:
done